Heroes.ReplayParser/MPQFiles/ReplayInitData.cs

```diff
@@ -277,37 +277,54 @@ public static void Parse(Replay replay, byte[] buffer)
                         reader.Read(32); // m_commanderLevel - So far, always 0
                     }
 
-					if (reader.ReadBoolean() && userID.HasValue) // m_hasSilencePenalty
+                    if (reader.ReadBoolean() && userID.HasValue) // m_hasSilencePenalty
                         replay.ClientListByUserID[userID.Value].IsSilenced = true;
 
                     if (replay.ReplayBuild >= 61718 && reader.ReadBoolean() && userID.HasValue) // m_hasVoiceSilencePenalty
                         replay.ClientListByUserID[userID.Value].IsVoiceSilence = true;
 
-					if(replay.ReplayBuild >= 66977 && reader.ReadBoolean() && userID.HasValue) // m_isBlizzardStaff
-						replay.ClientListByUserID[userID.Value].IsBlizzardStaff = true;
+                    if (replay.ReplayBuild >= 66977 && reader.ReadBoolean() && userID.HasValue) // m_isBlizzardStaff
+                        replay.ClientListByUserID[userID.Value].IsBlizzardStaff = true;
 
                     if (replay.ReplayBuild >= 69947 && reader.ReadBoolean() && userID.HasValue) // m_hasActiveBoost
                         replay.ClientListByUserID[userID.Value].HasActiveBoost = true;
 
                     if (replay.ReplayVersionMajor >= 2)
-					{
-						Encoding.UTF8.GetString(reader.ReadBlobPrecededWithLength(9)); // m_banner
-						Encoding.UTF8.GetString(reader.ReadBlobPrecededWithLength(9)); // m_spray
-						Encoding.UTF8.GetString(reader.ReadBlobPrecededWithLength(9)); // m_announcerPack
-						Encoding.UTF8.GetString(reader.ReadBlobPrecededWithLength(9)); // m_voiceLine
-
-						// m_heroMasteryTiers
-						if(replay.ReplayBuild >= 52561)
-						{
-							var heroMasteryTiersLength = reader.Read(10);
-							for (var j = 0; j < heroMasteryTiersLength; j++)
-							{
-								var heroAttributeName = new string(BitConverter.GetBytes(reader.Read(32)).Select(k => (char)k).Reverse().ToArray()); // m_hero
-								reader.Read(8); // m_tier
-							}
-						}
-					}
-				}
+                    {
+                        string banner = Encoding.UTF8.GetString(reader.ReadBlobPrecededWithLength(9)); // m_banner
+                        if (!string.IsNullOrEmpty(banner))
```

Comment:
Please check if userID has a value.  I may be remembering wrong, but I'm pretty sure there are replay files where userID doesn't have a value

```suggestion
                        if (!string.IsNullOrEmpty(banner) && userID.HasValue)
```